homeassistant/helpers/debounce.py

```diff
@@ -118,8 +118,7 @@ async def _handle_timer_finish(self) -> None:
             # Schedule a new timer to prevent new runs during cooldown
             self._schedule_timer()
 
-    @callback
-    def async_shutdown(self) -> None:
+    async def async_shutdown(self) -> None:
```

Comment:
In a follow-up PR, we could then add task cancellation (default to false) with optional await
```python
    async def async_shutdown(self) -> None:
        """Cancel any scheduled call, and prevent new runs."""
        self._shutdown_requested = True
        self.async_cancel()
        if cancel_existing_run:
            self._task.cancel()
        if wait_existing_run:
            await asyncio.wait((self.task,))
```

Reply:
I'd try to avoid `await asyncio.wait((self._task,))` since it creates a task

Reply:
Maybe this will be better?
```python
    async def async_shutdown(
        self,
        *,
        cancel_existing_run: bool | None = None,
        wait_existing_run: bool | None = None
    ) -> None:
        """Cancel any scheduled call, and prevent new runs."""
        self._shutdown_requested = True
        self.async_cancel()
        if cancel_existing_run:
            self._task.cancel()
        if wait_existing_run and not self._task.done():
            await asyncio.wait((self._task,))
```